caproto/curio/client.py

```diff
@@ -44,7 +44,7 @@ def __init__(self, circuit):
             raise RuntimeError("must await create_connection() first")
         bytes_received = await self.socket.recv(32768)
         if not len(bytes_received):
-            raise ca.DisconnectedCircuit()
```

Comment:
should this return here?  not sure throwing 0 bytes into circuit is correct now that the circuit thinks it is disconnected.